homeassistant/components/zone/__init__.py

```diff
@@ -95,79 +121,227 @@ def async_active_zone(hass, latitude, longitude, radius=0):
+        CONF_LONGITUDE: hass.config.longitude,
+        CONF_RADIUS: DEFAULT_RADIUS,
+        CONF_ICON: ICON_HOME,
+        CONF_PASSIVE: False,
+    }
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry
+) -> bool:
     """Set up zone as config entry."""
-    entry = config_entry.data
-    name = entry[CONF_NAME]
-    zone = Zone(
-        hass,
-        name,
-        entry[CONF_LATITUDE],
-        entry[CONF_LONGITUDE],
-        entry.get(CONF_RADIUS, DEFAULT_RADIUS),
-        entry.get(CONF_ICON),
-        entry.get(CONF_PASSIVE, DEFAULT_PASSIVE),
-    )
-    zone.entity_id = async_generate_entity_id(ENTITY_ID_FORMAT, name, None, hass)
-    hass.async_create_task(zone.async_update_ha_state())
-    hass.data[DOMAIN][slugify(name)] = zone
+    storage_collection = cast(ZoneStorageCollection, hass.data[DOMAIN])
+
+    data = dict(config_entry.data)
+    data.setdefault(CONF_PASSIVE, DEFAULT_PASSIVE)
+    data.setdefault(CONF_RADIUS, DEFAULT_RADIUS)
+
+    await storage_collection.async_create_item(data)
+
+    hass.async_create_task(hass.config_entries.async_remove(config_entry.entry_id))
+
     return True
 
 
-async def async_unload_entry(hass, config_entry):
-    """Unload a config entry."""
-    zones = hass.data[DOMAIN]
-    name = slugify(config_entry.data[CONF_NAME])
-    zone = zones.pop(name)
-    await zone.async_remove()
+async def async_unload_entry(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry
+) -> bool:
+    """Will be called once we remove it."""
     return True
+
+
+class Zone(entity.Entity):
+    """Representation of a Zone."""
+
+    def __init__(self, config: Dict, editable: bool):
+        """Initialize the zone."""
+        self._config = config
+        self._editable = editable
+        self._attrs: Optional[Dict] = None
+        self._generate_attrs()
+
+    @property
+    def state(self) -> str:
+        """Return the state property really does nothing for a zone."""
+        return "zoning"
+
+    @property
+    def name(self) -> str:
+        """Return name."""
+        return cast(str, self._config[CONF_NAME])
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return unique ID."""
+        return self._config.get(CONF_ID)
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon if any."""
+        return self._config.get(CONF_ICON)
+
+    @property
+    def state_attributes(self) -> Optional[Dict]:
+        """Return the state attributes of the zone."""
+        return self._attrs
+
+    async def async_update_config(self, config: Dict) -> None:
+        """Handle when the config is updated."""
+        self._config = config
+        self._generate_attrs()
+        self.async_write_ha_state()
+
+    @callback
+    def _generate_attrs(self) -> None:
+        """Generate new attrs based on config."""
+        self._attrs = {
+            ATTR_HIDDEN: True,
```

Comment:
Is this a GUI only attribute? Should we remove this, since we're removing it in general?